org.uqbar.project.wollok.lib/src/wollok/lang/WDictionary.xtend

```diff
@@ -0,0 +1,60 @@
+package wollok.lang
+
+import java.util.Collection
+import java.util.Map
+import java.util.Map.Entry
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.uqbar.project.wollok.interpreter.api.WollokInterpreterAccess
+import org.uqbar.project.wollok.interpreter.core.WollokObject
+import org.uqbar.project.wollok.interpreter.nativeobj.JavaWrapper
+
+import static extension org.uqbar.project.wollok.interpreter.nativeobj.WollokJavaConversions.*
+import static extension org.uqbar.project.wollok.lib.WollokSDKExtensions.*
+
+class WDictionary implements JavaWrapper<Map> {
+	
+	val WollokObject wollokInstance
+	@Accessors var Map wrapped
+	protected extension WollokInterpreterAccess = new WollokInterpreterAccess
+		 
+	new(WollokObject o) {
+		wollokInstance = o
+		wrapped = newHashMap
+	}
+	
+	def clear() { wrapped.clear }
+	
+	def basicGet(WollokObject key) { wrapped.get(getInternalKey(key)) }
+	
+	def put(WollokObject key, WollokObject value) { wrapped.put(key, value) }
+
+	def void remove(WollokObject key) {
+		wrapped.remove(getInternalKey(key))
+	}
+	
+	private def Object getInternalKey(WollokObject key) {
```

Comment:
I originally used wrapped.get(key), and wrapped.remove(key) but two Strings as WollokObjects are never equals. So get method always returned null.
I was stuck until I saw WList implementation of remove.
I'm not sure if there is a possibility of changing WollokObject definition of equals / hash to be like wollokEquals.